### PR TITLE
fix: error autocomplete when authoring_organizations is empty

### PR DESCRIPTION
**Pull request by binodpant**
JIRA: https://openedx.atlassian.net/browse/ENT-5520
Also fixes

- [x] fix unique key error
- [x]  gitignore edits to ignore coverage files

**Merge checklist:**
- [x] Ensure your commit message follows the semantic-release conventional commit message format
- [x] Once CI is passing, verify the package versions that Lerna will increment to in the Github Action CI workflow logs.
    - *Note*: This may be found in the "Preview Updated Versions (dry run)" step in the Github Action CI workflow logs.

**Post merge:**
- [ ] Verify Lerna created a release commit (e.g., ``chore(release): publish``) that incremented versions in relevant package.json and CHANGELOG files, and created [Git tags](https://github.com/edx/frontend-enterprise/tags) for those versions.
- [ ] Run the ``Publish from package.json`` Github Action [workflow](https://github.com/edx/frontend-enterprise/actions/workflows/publish-from-package.yml) to publish these new package versions to NPM.
    - This may be triggered by clicking the "Run workflow" option for the ``master`` branch.
- [ ] Verify the new package versions were published to NPM (i.e., ``npm view <package_name> versions --json``).
    - *Note*: There may be a slight delay between when the workflow finished and when NPM reports the package version as being published. If it doesn't appear right away in the above command, try again in a few minutes.

### CODE REVIEW

File: packages/catalog-search/src/SearchSuggestions.jsx [hunk 1]
```diff
@@ -15,37 +15,42 @@ const SearchSuggestions = ({ autoCompleteHits, enterpriseSlug, handleViewAllClic
...
+              <SearchSuggestionItem
+                key={hit.title}
+                url={getLinkToCourse(hit)}
+                /* eslint-disable-next-line no-underscore-dangle */
+                highlightedTitle={hit._highlightResult.title.value}
+                authoringOrganization={hit.key && hit.key.split('+')[0]}
+                title={hit.title}
+              />
+            ))
+        }
```
Comment (binodpant) [inline packages/catalog-search/src/SearchSuggestions.jsx:30]: this is just formatting fixes

File: packages/catalog-search/src/SearchSuggestions.jsx [hunk 2]
```diff
@@ -15,37 +15,42 @@ const SearchSuggestions = ({ autoCompleteHits, enterpriseSlug, handleViewAllClic
...
-                          authoringOrganization={hit.authoring_organizations && hit.authoring_organizations[0].key}
-                          title={hit.title}
-                          programType={hit.program_type}
-                        />
-                      ))
-                  }
+          autoCompleteHits.filter(hit => hit.content_type === 'program')
+            .slice(0, MAX_NUM_SUGGESTIONS)
+            .map((hit) => {
+              const orgFound = hit.authoring_organizations && hit.authoring_organizations.length > 0;
```
Comment (binodpant) [inline packages/catalog-search/src/SearchSuggestions.jsx:40]: this is the actual fix

File: packages/catalog-search/src/tests/SearchSuggestions.test.jsx [hunk 3]
```diff
@@ -45,6 +61,20 @@ describe('<SeachSuggestions />', () => {
     expect(screen.getByText('View all results')).not.toBeNull();
   });
 
+  test('renders no errors when no authoring orgs found for programs data', () => {
+    renderWithRouter(<SearchSuggestions
```
Comment (binodpant) [inline packages/catalog-search/src/tests/SearchSuggestions.test.jsx:65]: verified this tesst actually does fail the same way as reported inthe ticket, in the absence of the fix :)

File: packages/catalog-search/src/SearchSuggestions.jsx [hunk 4]
```diff
@@ -15,37 +15,42 @@ const SearchSuggestions = ({ autoCompleteHits, enterpriseSlug, handleViewAllClic
...
+              return (
+                <SearchSuggestionItem
+                  key={hit.title}
```
Comment (binodpant) [inline packages/catalog-search/src/SearchSuggestions.jsx:43]: this fixes the unique key warnings I was seeing in test logs

File: packages/catalog-search/src/SearchSuggestions.jsx (hunk 2 above)
Comment (adamstankiewicz) [inline packages/catalog-search/src/SearchSuggestions.jsx:40]: nit: `const orgFound = hit.authoring_organizations?.length > 0;`

And maybe `hasAuthoringOrg` vs. `orgFound` to be a bit more descriptive?

File: packages/catalog-search/src/SearchSuggestions.jsx [hunk 5]
```diff
@@ -15,37 +15,42 @@ const SearchSuggestions = ({ autoCompleteHits, enterpriseSlug, handleViewAllClic
...
+                  url={getLinkToProgram(hit)}
+                  /* eslint-disable-next-line no-underscore-dangle */
+                  highlightedTitle={hit._highlightResult.title.value}
+                  authoringOrganization={orgFound ? hit.authoring_organizations[0].key : undefined}
```
Comment (adamstankiewicz) [inline packages/catalog-search/src/SearchSuggestions.jsx:47]: Alternatively, you might also do this in a more shorthand way as well:

`authoringOrganization ={hit.authoring_organizations.shift()?.key}`

Comment (binodpant) [reply packages/catalog-search/src/SearchSuggestions.jsx:47]: cool tip thanks!

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**